tests/core/test_library.py

```diff
@@ -419,3 +419,71 @@ def test_core_search_with_handles_legacy_backend(self):
         self.backend.library.search.return_value.get.side_effect = TypeError
         self.core.library.search(query={'any': ['a']}, exact=True)
         # We are just testing that this doesn't fail.
+
+
+@mock.patch('mopidy.core.library.logger')
+class BackendFailuresCoreLibraryTest(unittest.TestCase):
+
+    def setUp(self):  # noqa: N802
+        dummy_root = Ref.directory(uri='dummy:directory', name='dummy')
+
+        self.library = mock.Mock(spec=backend.LibraryProvider)
+        self.library.root_directory.get.return_value = dummy_root
+
+        self.backend = mock.Mock()
+        self.backend.actor_ref.actor_class.__name__ = 'DummyBackend'
+        self.backend.uri_schemes.get.return_value = ['dummy']
+        self.backend.library = self.library
+
+        self.core = core.Core(mixer=None, backends=[self.backend])
+
+    def test_browse_backend_get_root_exception_gets_ignored(self, logger):
+        # Might happen if root_directory is a property for some weird reason.
+        self.library.root_directory.get.side_effect = Exception
+        self.assertEqual([], self.core.library.browse(None))
+        logger.exception.assert_called_with(mock.ANY, 'DummyBackend')
+
+    def test_browse_backend_browse_uri_exception_gets_through(self, logger):
+        # TODO: is this behavior desired?
```

Comment:
Can we fix this now? If exception's not part of the API here, let's not let it through.